tb_plugin/torch_tb_profiler/profiler/loader.py

```diff
@@ -40,33 +41,54 @@ def _parse(self):
                 continue
 
             worker = match.group(1)
-            workers.append((worker, path))
+            span = match.group(2)
+            if span:
+                # remove the starting dot (.)
+                span = span[1:]
+                spans.append(span)
 
-        for worker, path in sorted(workers):
+            workers.append((worker, span, path))
+
+        spans.sort()
+
+        for worker, span, path in sorted(workers):
             try:
-                data = RunProfileData.parse(self.run.run_dir, worker, path, self.caches)
-                self.run.profiles[worker] = data
+                # convert the span timestamp to the index.
+                span_index = None if span is None else spans.index(span)
+                data = RunProfileData.parse(self.run.run_dir, worker, span_index, path, self.caches)
+                self.run.profiles[(worker, span)] = data
             except Exception as ex:
                 logger.warning("Failed to parse profile data for Run %s on %s. Exception=%s",
                                self.run.name, worker, ex, exc_info=True)
 
     def _process(self):
+        spans = self.run.get_spans()
+        if spans is None:
+            self.has_communication = self._process_profiles(self.run.profiles.values())
+        else:
+            self.has_communication = {}
+            for span in spans:
+                profiles = self.run.get_profiles(span)
+                self.has_communication[span] = self._process_profiles(profiles)
+
+    def _process_profiles(self, profiles):
+        has_communication = True
```

Comment:
Add an check : 'if not profiles: return False'?

Reply:
we should not add the check here. if it happens, there must be some bugs at other place.